prov/efa/src/efa_device.c

```diff
@@ -97,6 +97,11 @@ int efa_device_construct(struct efa_device *efa_device,
 		goto err_close;
 	}
 
+	if (efa_device->ibv_attr.vendor_part_id == 0xefa2) {
```

Comment:
I think we should use an array for different `vendor_part_id`. e.g.

```
static int use_device_rdma[] = {
      [0] = 0,
      [1] = 0,
      [2] = 1};

rxr_env.use_device_rdma = use_device_rdma[efa_device->ibv_attr.vender_pard_id - 0xefa0);
```
Make it earier to add new device and change decision for different id.

Reply:
Thank you! Updated.

Reply:
I'd skip the table. Just set it to 0 if the device id is 0xefa0 or 0xefa1. We're not going to go back on the default, so just do the future proof thing.

Reply:
The latest revision set `use_device_rdma` to 0 when the efa device id is < 0xefa2. There is a corner case that user could `FI_EFA_USE_DEVICE_RDMA=1` on p4d, which has `0xefa0`. So I add a check for `fi_get_param`. But I think mixing the env with the runtime default may not be the best practice. I am open to suggestions here.